main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidator.java

```diff
@@ -32,56 +34,99 @@
  * <ul>
  *   <li>{@link StopTimeTimepointWithoutTimesNotice} - a timepoint does not specifies arrival_time
  *       or departure_time
+ *   <li>{@link MissingTimepointValueNotice} - value for {@code stop_times.timepoint} is missing
+ *   <li>{@link MissingTimepointColumnNotice} - field {@code stop_times.timepoint} is missing
  * </ul>
  */
 @GtfsValidator
-public class TimepointTimeValidator extends SingleEntityValidator<GtfsStopTime> {
+public class TimepointTimeValidator extends FileValidator {
+  private final GtfsStopTimeTableContainer stopTimes;
+
+  @Inject
+  TimepointTimeValidator(GtfsStopTimeTableContainer stopTimes) {
+    this.stopTimes = stopTimes;
+  }
 
   @Override
-  public void validate(GtfsStopTime stopTime, NoticeContainer noticeContainer) {
-    if (!isTimepoint(stopTime)) {
+  public void validate(NoticeContainer noticeContainer) {
+    boolean useTimepointColumn = stopTimes.hasColumn(GtfsStopTimeTableLoader.TIMEPOINT_FIELD_NAME);
+    if (!useTimepointColumn) {
+      // legacy datasets do not use timepoint column in stop_times.txt as a result:
+      // - this should be flagged;
+      // - but also no notice regarding the absence of arrival_time or departure_time should be
+      // generated
+      noticeContainer.addValidationNotice(new MissingTimepointColumnNotice());
       return;
     }
-    if (!stopTime.hasArrivalTime()) {
-      noticeContainer.addValidationNotice(
-          new StopTimeTimepointWithoutTimesNotice(
-              stopTime.csvRowNumber(),
-              stopTime.tripId(),
-              stopTime.stopSequence(),
-              String.format(GtfsStopTimeTableLoader.ARRIVAL_TIME_FIELD_NAME)));
-    }
-    if (!stopTime.hasDepartureTime()) {
-      noticeContainer.addValidationNotice(
-          new StopTimeTimepointWithoutTimesNotice(
-              stopTime.csvRowNumber(),
-              stopTime.tripId(),
-              stopTime.stopSequence(),
-              GtfsStopTimeTableLoader.DEPARTURE_TIME_FIELD_NAME));
+    for (GtfsStopTime stopTime : stopTimes.getEntities()) {
+      if (!stopTime.hasTimepoint()) {
+        noticeContainer.addValidationNotice(new MissingTimepointValueNotice(stopTime));
+      }
+      if (isTimepoint(stopTime)) {
+        if (!stopTime.hasArrivalTime()) {
+          noticeContainer.addValidationNotice(
+              new StopTimeTimepointWithoutTimesNotice(
+                  stopTime, String.format(GtfsStopTimeTableLoader.ARRIVAL_TIME_FIELD_NAME)));
+        }
+        if (!stopTime.hasDepartureTime()) {
+          noticeContainer.addValidationNotice(
+              new StopTimeTimepointWithoutTimesNotice(
+                  stopTime, GtfsStopTimeTableLoader.DEPARTURE_TIME_FIELD_NAME));
+        }
+      }
     }
   }
 
   private boolean isTimepoint(GtfsStopTime stopTime) {
-    return stopTime.timepoint().equals(GtfsStopTimeTimepoint.EXACT);
+    return stopTime.hasTimepoint() && stopTime.timepoint().equals(GtfsStopTimeTimepoint.EXACT);
```

Comment:
Actually because of the fact that `timepoint` has a default value we need to check `stopTime.hasTimepoint()`. See, if a `stop_time` has no value for timepoint or has value `1` for `timepoint`, `stopTime.timepoint()` return the same value `EXACT`. Using `stopTime.hasTimepoint()` makes it possible to distinguish these two situations.

Reply:
I see, in this case should we remove default value instead?

Reply:
I could see this happening after the specification is clarified to avoid making the behavior of the validator  contradictory/not in accordance with the specs. Thoughts @barbeau @isabelle-dr?

Reply:
Agreed to remove default value.

Reply:
>I see, in this case should we remove default value instead?

So if we remove `@DefaultValue("1")`, the new value of this field would be `GtfsStopTimeTimepoint.UNRECOGNIZED` if the timepoint column is missing or the field has an empty value in the CSV?

I'm still not sure that's what we want.

If we define a default, I think the actual behavior we would want is that the default is set to `EXACT` IFF the timepoint column isn't provided - this matches the original GTFS spec before the `timepoint` field was added for legacy datasets. 

Otherwise, IMHO, if the column exists but only some records have a value for `timepoint` field, then `GtfsStopTimeTimepoint.UNRECOGNIZED` seems appropriate to me for the records that are empty (because the original intent of adding timepoint column was to have producers specify `0` or `1` for every record to be unambiguous about whether a record was a timepoint).

It also might be confusing as to why this value is missing a default when other fields in the spec that have a `empty` value do have a default - so we should explain in comments why.

Related - I was curious to see how other defaults were handled, but it looks like `@DefaultValue` is only sparingly used currently - in `GtfsStopTimeSchema` for timepoint and in both `GtfsStopTimeSchema` and `GtfsRouteSchema` for `continuousPickup` and `continuousDropOff`, and in `GtfsRouteSchema` for `routeColor` and `routeTextColor`. 

It looks like the defaults for  `continuousPickup` and `continuousDropOff` aren't referenced anywhere, but the defaults for `routeColor` and `routeTextColor` are used for validating contrast.

But - it looks like defaults are handled as an enum value for other enums?

Take a look at `GtfsWheelchairBoardingEnum` and `GtfsTripSchema` for trips.txt. The spec says:

>Indicates wheelchair accessibility. Valid options are:
>
>* 0 or empty - No accessibility information for the trip.
>* 1 - Vehicle being used on this particular trip can accommodate at least one rider in a wheelchair.
>* 2 - No riders in wheelchairs can be accommodated on this trip.

If I'm understanding the code correctly it looks like if the column or field value is empty/missing, `GtfsWheelchairBoarding.UNRECOGNIZED` is returned. But from reading the spec, it seems like this should actually be `GtfsWheelchairBoarding.UNKNOWN`? So should this field be annotated with `@DefaultValue("0")`?

Same with `bikes_allowed`.

IMHO this is another reason to have test CSV data, as it makes it clearer what the expected behavior of the validator is in case of defaults/empty values/columns.

Reply:
I can see there are a couple of potential problems. 

With timepoint, if we're using different default value in this validation rule and for everything else, this will create confusion and potential bugs. We can keep it this way and then remove `DefaultValue` from `timepoint` once specification is fixed, on other hand this rule contradicts to specification and we might want to change specification first before changing the validator. I'm fine with either solution in order to unblock the PR.

For other default values, it seems, that default variable type value is used (e.g. 0 for integers), so `@DefaultValue` is used only when a default value for a field is unusual, we might want to specify `DefaultValue` explicitly everywhere.

Reply:
>For other default values, it seems, that default variable type value is used (e.g. 0 for integers),

@asvechnikov2 I was trying to trace the execution for this too. Where do you see this assignment?

Reply:
I think it's this one https://github.com/MobilityData/gtfs-validator/blob/903e9af59f4b959d5186d890e4eee1bfa4fcb89b/processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/EntityImplementationGenerator.java#L134-L160

Reply:
@asvechnikov2 Ah, thanks! Looks like you're right. I just ran this through the debugger with two different datasets to confirm:
* [hart_timepoint_header_only.zip](https://github.com/MobilityData/gtfs-validator/files/7506789/hart_timepoint_header_only.zip)
* [hart_no_timepoint_column.zip](https://github.com/MobilityData/gtfs-validator/files/7506790/hart_no_timepoint_column.zip)

This code is what auto-generates the `GtfsStopTime.DEFAULT_TIMEPOINT` variable value. 

So the current behavior is:
* If there is no `@DefaultValue` annotation, if the column is missing or the row has an empty value for an enum then the default value of `0` is used. 
* If there is a `@DefaultValue` annotation, then then provided default value is used.

So the default enum value is the value defined for `0` - for `GtfsStopTimeTimepoint` this is `APPROXIMATE(0)`. This is similar to protocol buffer bindings behavior. I agree that explicitly defining the default enum value everywhere via `@DefaultValue` would be a good best practice to avoid confusion, especially for newcomers adding new fields where is isn't obvious that the `0` enum value is the de facto default.

Back to the original question - if we remove `@DefaultValue("1")`, the new value of `timepoint` field would be `GtfsStopTimeTimepoint.APPROXIMATE` if the timepoint column is missing or the field has an empty value in the CSV. We don't want this because this either because of legacy datasets from the original GTFS spec before the timepoint field existed (because all those records should be considered `EXACT`.

IMHO the best solution is still having the default is set to `EXACT` IFF the timepoint column isn't provided, and otherwise have the default set to `UNRECOGNIZED` for empty. I think the 2nd best solution is to leave the ``@DefaultValue("1")`, because that aligns with the legacy GTFS datasets and possible interpretations of the current GTFS spec.

Reply:
nit: I'd add a comment to clarify the behaviour, here and probably in the class docstring as well.